www/js/diary/current.js

```diff
@@ -110,115 +173,95 @@
-    map.removeLayer(marker);
+  var getLocalIncidents = function() {
+    db.getAllMessages(MANUAL_INCIDENT).then(function(incidents) {
+      console.log("Incidents stored locally", incidents);
+      addIncidents(incidents, _map);
+    });
   };
 
-  var addmarker = function(map, ts, llng) {
-      console.log("both", llng);      
-      console.log("called addmarker with map = "+map);
-      var latlng = L.latLng(llng);
-      var m = L.circleMarker(latlng);
-      var safe_suck_cancel_actions = [{text: "<font size='+5'>&#x263B;</font>",
-                                     action: addSafe},
-                                    {text: "<font size='+5'>&#x2639;</font>",
-                                     action: addSuck},
-                                    {text: "Cancel",
-                                     action: cancelTempEntry}];
-    Logger.log("About to call ionicActionSheet.show");
-    $ionicActionSheet.show({titleText: "lat: "+latlng.lat.toFixed(6)+
-              ", lng: " + latlng.lng.toFixed(6),
-          cancel: function() {
-            cancelTempEntry(m, ts, map);
-          },
-          buttons: safe_suck_cancel_actions,
-          buttonClicked: function(index, button) {
-            button.action(m, ts, map);
-            Logger.log("Clicked button "+button.text+" at index "+index);
-            return true;
-          }
+  var getServerIncidents = function() {
+      console.log(incidentServerCalldata);
+      $http.post("https://e-mission.eecs.berkeley.edu/result/heatmap/incidents/local_date", incidentServerCalldata).then(function(res){
+      console.log(res);
+      if(res.data.incidents.length > 0) {
+        addIncidents(res.data.incidents, _map);
+      }
+    }, function(error){
+      console.log("Error when getting incidents");
+      console.log(error);
     });
-      
   };
-  
-  $scope.parse = function() {
-    console.log("About to get map"); 
-    var _map;
-    leafletData.getMap('current').then(function(map) {
-      console.log("success, got map "+ map);
-      _map = map;
-      return db.getAllSensorData(BACKGROUND_LOCATION);
-    }).then(function(result) {
+
+  var marker;
+  $scope.showIncidentSheet = function() {
+    db.getAllSensorData(BACKGROUND_LOCATION).then(function(result) {
             both = [result[0].data.latitude, result[0].data.longitude];
             var ts = result[0].data.ts;
-            addmarker(_map, ts, both);
+            var latlng = L.latLng(both);
+            $scope.features = [];
+            marker = L.circleMarker(latlng);
+            console.log(marker);
+            PostTripManualMarker.showSheet($scope.features, latlng, ts, marker, _map);
     })
     .catch(function(error) {
       console.log("error while getting map current from leafletData");
     });
   };
 
+  $scope.$watch('features', function(newVal, oldVal){
+    if(newVal.length === 1) {
+      addIncidentLayer(newVal[0].properties.stress, marker, _map);
```

Comment:
`showSheet` does not add the marker to the map, it just stores the data locally.